app/utils/reflect.js

```diff
@@ -0,0 +1,12 @@
+'use strict'
+
+module.exports = {
+  reflect: promise => {
+    return promise.then(function (v) {
+      return {v: v, status: 'resolved'}
+    },
+    function (e) {
+      return {e: e, status: 'rejected'}
+    })
+  }
+}
```

Comment:
Would this be too verbose?
```js
module.exports = {
  reflect: promise => {
    return promise.then(result => ({ result, status: 'resolved' }),
      error => ({ error, status: 'rejected' })
    )
  }
}
```

Reply:
I think it probably is :) looking at it

Reply:
What about 

```js
'use strict'

module.exports = {
  reflect: promise => {
    return promise.then(result => {
      return { result, status: 'resolved' }
    },
    error => {
      return { error, status: 'rejected' }
    })
  }
}
```

Feel free to totally ignore me though

Reply:
Agree that looks better, I've updated the code with that implementation.